TRRandomizerCore/Globalisation/G11N.cs

```diff
@@ -8,89 +8,100 @@ namespace TRRandomizerCore.Globalisation
-                _languageMap[language] = null;
-                if (!language.IsHybrid)
+                IGameStrings strings;
+                if (!language.IsHybrid && (strings = LoadLanguage(language, _game)) != null)
                 {
+                    _languageMap[language] = strings;
                     _realLanguages.Add(language);
                 }
             }
         }
 
-        public Language GetLanguage(string tag)
+        public static List<Language> GetSupportedLanguages(G11NGame game)
+        {
+            List<Language> languages = new List<Language>();
+            foreach (Language language in _definedLanguages)
+            {
+                if (language.IsHybrid || LoadLanguage(language, game) != null)
+                {
+                    languages.Add(language);
+                }
+            }
+            return languages;
+        }
+
+        public static Language GetLanguage(string tag)
         {
             tag = tag.ToUpper();
-            return _languageMap.Keys.ToList().Find(l => l.Tag.ToUpper().Equals(tag));
+            return _definedLanguages.Find(l => l.Tag.ToUpper().Equals(tag));
         }
 
-        public GameStrings GetGameStrings(string tag, G11NGame game)
+        public IGameStrings GetDefaultGameStrings()
         {
-            return GetGameStrings(GetLanguage(tag), game);
+            return GetGameStrings(GetLanguage(Language.DefaultTag));
         }
 
-        public GameStrings GetGameStrings(Language language, G11NGame game)
+        public IGameStrings GetGameStrings(string tag)
+        {
+            return GetGameStrings(GetLanguage(tag));
+        }
+
+        public IGameStrings GetGameStrings(Language language)
         {
             if (!_languageMap.ContainsKey(language))
             {
                 throw new KeyNotFoundException(string.Format("There is no language defined for {0} ({1})).", language.Name, language.Tag));
             }
 
-            if (_languageMap[language] == null && !language.IsHybrid)
-            {
-                try
-                {
-                    string path = string.Format(@"Resources\{0}\Strings\G11N\gamestrings_{1}.json", game.ToString(), language.Tag);
-                    _languageMap[language] = JsonConvert.DeserializeObject<GameStrings>(File.ReadAllText(path, Encoding.UTF8));
-                }
-                catch (FileNotFoundException)
-                {
-                    //A language for a game isn't supported
-                }
-            }
-
             return _languageMap[language];
         }
 
-        public GameStrings GetDefaultGameStrings(G11NGame game)
+        private static IGameStrings LoadLanguage(Language language, G11NGame game)
         {
-            return GetGameStrings(GetLanguage(Language.DefaultTag), game);
+            string path = string.Format(@"Resources\{0}\Strings\G11N\gamestrings_{1}.json", game.ToString(), language.Tag);
+            if (!File.Exists(path))
```

Comment:
Might like this more?:
```
string path = $@"Resources\{game}\Strings\G11N\gamestrings_{language.Tag}.json";
```

Alternatively there is `Path.Join` which will do correct path separators for current platform (not really relevant I suppose):
```
string path = Path.Join("Resources", game, "Strings", "G11N", $"gamestrings_{language.Tag}.json");
```